### PR TITLE
Unify form fields with database text fields

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Continuing on https://github.com/PrestaShop/PrestaShop/pull/35749, it's also needed to unify the forms with the database limits. More info below.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See that product description can now store much more data correctly.
| UI Tests          | 
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | 

### Info
- Changes all hardcoded numbers million times into a constant.
- Sets limits of MEDIUMTEXT fields on forms, so we can save more data. Made it everywhere where it made sense.
- As a side bonus, it allows to save much more data everywhere. Improves annoyances like https://github.com/PrestaShop/PrestaShop/issues/37229, where you can't save a legit product description with some HTML.

### Limits set
ps_category_lang	description
ps_category_lang	additional_description
ps_cms_lang	content
ps_manufacturer_lang	description
ps_manufacturer_lang	short_description
ps_product_lang	description
ps_supplier_lang	description

All others are either internal, legacy or using TextareaType that doesn't have limit option.

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `branch` should be `develop` or `8.2.x`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `How to test` shouldn&#039;t be empty. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#how-to-test))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

File: classes/CMSCategory.php [hunk 1]
```diff
@@ -23,7 +23,10 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
-class CMSCategoryCore extends ObjectModel
+
+use PrestaShopBundle\Form\Admin\Type\FormattedTextareaType;
+
+ class CMSCategoryCore extends ObjectModel
```
Comment (kpodemski) [inline classes/CMSCategory.php]: woopsie

Comment (Hlavtox) [reply classes/CMSCategory.php]: Expected much worse, fixed :D

**Review by ShaiMagal — APPROVED** [review]

File: src/PrestaShopBundle/Form/Admin/Type/FormattedTextareaType.php [hunk 2]
```diff
@@ -40,14 +40,26 @@ class FormattedTextareaType extends TranslatorAwareType
...
     public const LIMIT_LONGTEXT_UTF8 = 1431655764;
 
     /**
      * Max size of UTF-8 MB4 content in MySQL text columns
+     *
+     * To calculate these, we start from the raw byte limit of these fieds:
+     * TINYTEXT 255
+     * TEXT 65535
+     * MEDIUMTEXT 16777215
+     * LONGTEST 4294967295
```
Comment (SharakPL) [inline src/PrestaShopBundle/Form/Admin/Type/FormattedTextareaType.php:62]: LONGTEXT 😉
fields 😉

**Review by SharakPL — APPROVED** [review]
Small mistake in the comments. Otherwise LGTM

File: src/PrestaShopBundle/Form/Admin/Type/FormattedTextareaType.php (hunk 2 above)
Comment (Hlavtox) [reply src/PrestaShopBundle/Form/Admin/Type/FormattedTextareaType.php:62]: I will change it in the next PR so I don't loose approvals :D

**Review by florine2623 — APPROVED** [review]
Hello @Hlavtox ,

The issue https://github.com/PrestaShop/PrestaShop/issues/37229 is fixed now. I can save the long description successfully. I have no more issue telling me that the field cannot be longer that XXX characters.

I followed the steps to reproduce of the above issue. In the FO, the description is well saved :

Screenshot 2024-11-13 at 10 56 46

The automated tests are green : https://github.com/florine2623/testing_pr/actions/runs/11813760647 ✅ 

It is QA ✅

**Pull request merged**